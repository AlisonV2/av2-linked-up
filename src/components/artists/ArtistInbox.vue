<template>
  <div class="row artist-inbox">
    <div class="col-12" v-for="message in messages" :key="message.project">
      <router-link
        :to="{ name: 'MessageDetails', params: { id: message.project } }"
      >
        <div class="message-item">Conversation with : {{ message.client }}</div>
      </router-link>
    </div>
  </div>
</template>

<script>
import * as Sentry from '@sentry/vue';

/**
 * @exports ArtistInbox
 * @type {Component}
 * @requires Sentry
 */
export default {
  name: 'ArtistInbox',
  data() {
    return {
      messages: [],
    };
  },
  created() {
    this.$store
      .dispatch('getArtistMessages')
      .then(() => {
        this.messages = this.$store.getters.getArtistMessages;
      })
      .catch((err) => {
        Sentry.captureException(err);
      });
  },
};
</script>

<style lang="scss">
.message-item {
  color: $dark;
  border: 2px solid #dedede;
  background-color: #f1f1f1;
  border-radius: 5px;
  padding: 10px;
  margin: 10px 0;
}
</style>
