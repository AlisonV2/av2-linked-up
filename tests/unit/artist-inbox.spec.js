import { shallowMount, RouterLinkStub } from '@vue/test-utils';
import { createStore } from 'vuex';
import ArtistInbox from '@/components/artists/ArtistInbox.vue';

const actions = {
  getArtistMessages: jest.fn(),
};

const getters = {
  getArtistMessages: jest.fn(),
};

const store = createStore({
  actions,
  getters
});

const config = {
  data() {
    return {
      messages: []
    }
  },
  global: {
    plugins: [store],
    stubs: {
      'router-link': RouterLinkStub,
    }
  }
}

/**
 * @module ArtistInboxTest
 */
describe('ArtistInbox.vue', () => {
  it('Check if row class is relevant', () => {
    const component = shallowMount(ArtistInbox, config);
    const row = component.find('.artist-inbox')
    expect(row.exists()).toBe(true);
  });
  it('Check if getArtistMessages action is called', async () => {
    const component = shallowMount(ArtistInbox, config);
    ArtistInbox.created.call(component.vm);
    await component.vm.$nextTick();
    expect(actions.getArtistMessages).toHaveBeenCalled;
  });
  it('Check if getArtistMessages getter is called', async () => {
    const component = shallowMount(ArtistInbox, config);
    ArtistInbox.created.call(component.vm);
    await component.vm.$nextTick();
    expect(getters.getArtistMessages).toHaveBeenCalled;
  });
});
