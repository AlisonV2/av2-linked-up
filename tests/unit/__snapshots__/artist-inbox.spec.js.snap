// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistInbox.vue ArtistInbox Snapshot 1`] = `
<div
  class="row artist-inbox"
>
  
  
</div>
`;
